packages/core/src/external/carequality/ihe-gateway-v2/ihe-gateway-v2-async.ts

```diff
@@ -28,16 +31,26 @@ export class IHEGatewayV2Async extends IHEGatewayV2 {
     cxId: string;
   }): Promise<void> {
     const lambdaClient = makeLambdaClient(Config.getAWSRegion());
-    const params = { patientId, cxId, pdRequestGatewayV2 };
-    // intentionally not waiting
-    lambdaClient
-      .invoke({
-        FunctionName: iheGatewayV2OutboundPatientDiscoveryLambdaName,
-        InvocationType: "Event",
-        Payload: JSON.stringify(params),
-      })
-      .promise()
-      .catch(processAsyncError("Failed to invoke iheGatewayV2 lambda for patient discovery"));
+    const { gateways, ...rest } = pdRequestGatewayV2;
+    const gatewayChunks =
+      gateways.length > MAX_GATEWAYS_BEFORE_CHUNK
+        ? chunk(gateways, Math.ceil(gateways.length / 2))
```

Comment:
This will not guarantee each chunk has `MAX_GATEWAYS_BEFORE_CHUNK` elements.

I see why we don't want to just use `MAX_GATEWAYS_BEFORE_CHUNK` as chunk size, otherwise we can end up w/ a chunk of 100+ elements and another one with a handful.

It should be simple to just determine the size of each chunk considering the max elements we can have on each chunk:

```ts
const chunks = Match.ceil(gateways.length / MAX_GATEWAYS_PER_CHUNK);
const chunkSize = Match.ceil(gateways.length / chunks);
const gatewayChunks = chunk(gateways, chunkSize);
...
```

Reply:
Got it. Added this